core/src/main/java/io/cucumber/core/api/TypeRegistryConfigurer.java

```diff
@@ -6,8 +6,12 @@
 
 /**
  * The type registry configurer allows to configure a new type registry and the locale.
+ * @deprecated Please use annotation based configuration.
+ * See <a href="https://github.com/cucumber/cucumber-jvm/blob/master/examples/java-calculator/src/test/java/io/cucumber/examples/java/ShoppingSteps.java">Annotation based example</a>
+ * See <a href="https://github.com/cucumber/cucumber-jvm/blob/master/examples/java8-calculator/src/test/java/io/cucumber/examples/java8/ShoppingSteps.java">Lambda based example</a>
  */
-@API(status = API.Status.STABLE)
+@API(status = API.Status.DEPRECATED)
+@Deprecated
```

Comment:
There is a subtle difference between the mean of these two deprecations. The API Guardian deprecation documents that a method annotated this way be removed at any time. The java deprecation makes no such statements. 

Since we only remove deprecated methods at major release we should still use the `Status.Stable` here.